src/common/ast-path.js

```diff
@@ -110,6 +110,22 @@ class AstPath {
     return result;
   }
 
+  /**
+   * @param {() => void} callback
+   * @param {(error: any) => void} onCaught
+   * @internal Unstable API. Don't use in plugins for now.
+   */
+  try(callback, onCaught) {
+    const { stack } = this;
+    const { length } = stack;
+    try {
+      callback();
+    } catch (error) {
+      stack.length = length;
+      onCaught(error);
+    }
+  }
```

Comment:
Use error looks usual too me, but  how about

```js
try(callback) {
  callback();
} finally {
  stack.length = length;
 }
```

so we can use 

```js
try {
	path.try(() => {/* ... */})	
} catch (error) {}
```

Reply:
Wait, if `callback` callParent, restore `.length` is not right.

Reply:
I used `caught` because this thing we catch here logically is something expected, not an error.

> Wait, if callback callParent, restore .length is not right

Oh, you're right!

Reply:
> but how about

Looks interesting, I'll try it.

Reply:
Are you familiar with `CanvasRenderingContext2D.{save, restore}()`?

```js
path.save();
try {
  iterateCallArgumentsPath(...)
} catch (caught) {
  path.restore();
  // ....
}
```

WDYT? 😄

Reply:
`path.try` in its current form matches the style of other methods of `AstPath`.

Reply:
OR

```js
const [error, result]= path.try(() => {/* ... */});

if (error) {
  ...
}
```

Reply:
I like the current form more for its simplicity and because it makes you handle the error, you can't ignore it. But feel free to change it if you want.